src/renderers/shaders/ShaderChunk/envmap_fragment.glsl.js

```diff
@@ -26,7 +26,7 @@ export default /* glsl */`
 
 	#ifdef ENVMAP_TYPE_CUBE
 
-		vec4 envColor = textureCube( envMap, vec3( flipEnvMap * reflectVec.x, reflectVec.yz ) );
+		vec4 envColor = textureCube( envMap, envMapRotation * vec3( flipEnvMap * reflectVec.x, reflectVec.yz ) );
```

Comment:
Every pixel, for every frame? I would not support this -- even if it were wrapped in a #DEFINE.

Correct your envmap once before you use it.

Reply:
> Every pixel, for every frame? I would not support this

I agree that doing this multiplication so often is probably overkilled. Rotating the background at any time is certainly not very useful.

> even if it were wrapped in a #DEFINE.

I don't understand why you would not accept this in a DEFINE. Could you explain me your concern?

> Correct your envmap once before you use it.

I implemented this workaround since a while and it's working.
But I'm not satisfied by this because it increases the complexity of my application and it increases the loading time, either on the server or on the client side. Both are annoying.

I cannot change the background assets in my application because it would break our support of the VRML and X3D standard, and/or the huge user backward compatibility we are trying to support.

Reading at the issue, I'm far to be the only guy having this issue ^^ I think everybody who would like to support some background standard specification which doesn't match with the threejs rigid CubeTexture orientation will have this issue.

---

I would recommend you to accept a DEFINE mechanism at least, and I can still spend time to implement it. But if you don't agree, I don't want to use more time in this direction ;-) Please keep me in touch!

Reply:
Even with a DEFINE you would be rotating the reflection vector every pixel, every frame.

>Correct your envmap once before you use it.
>>I implemented this workaround since a while and it's working.

Great!

Reply:
@WestLangley yeah, every pixel every frame with a DEFINE to make it a user's choice, not yours, or propose more efficient solution, or just don't comment.